hack/generator/pkg/codegen/testdata/Flattening/FlattenedPropertiesAreFlattened.json

```diff
@@ -0,0 +1,29 @@
+{
+    "$comment": "Check that flattened definitions are inlined into parent type",
```

Comment:
If you're planning on merging this, probably worth a big TODO in the comment saying that it doesn't actually generate correct code because of the `gojsonschema` point you mentioned above.

Reply:
agreed